internal/server/data/provideruser.go

```diff
@@ -93,29 +93,60 @@ func UpdateProviderUser(tx WriteTxn, providerUser *models.ProviderUser) error {
 	return handleError(err)
 }
 
-func listProviderUsers(tx ReadTxn, providerID uid.ID) ([]models.ProviderUser, error) {
+func ListProviderUsers(tx ReadTxn, providerID uid.ID, p *SCIMParameters) ([]models.ProviderUser, error) {
 	table := &providerUserTable{}
 	query := querybuilder.New("SELECT")
 	query.B(columnsForSelect(table))
 	query.B("FROM")
 	query.B(table.Table())
 	query.B("WHERE provider_id = ?", providerID)
+
```

Comment:
To prevent someone guessing an ID and querying for other orgs, I think we would also need to do something like this:

```
INNER JOIN providers ON provider_users.provider_id = providers.id AND providers.organization_id = ?
```

Right now that's not a risk because we get the providerID from the access key, so not from user input.

I was wrong about not needing organization_id in this case previously, so I thought I'd mention it.

Reply:
Good point, I've added this protection